subsquid/src/pabloProcessor.ts

```diff
@@ -0,0 +1,299 @@
+        quoteAsset.blockNumber = BigInt(ctx.block.height);
+
+        let tx = await get(ctx.store, PabloTransaction, ctx.event.id);
+        if (tx != undefined) {
+            throw new Error("Unexpected transaction in db");
+        }
+        tx = createTransaction(ctx, pool, who,
+            PabloTransactionType.ADD_LIQUIDITY,
+            Big(liquidityAddedEvt.baseAmount.toString())
+                .div(Big(liquidityAddedEvt.quoteAmount.toString())).toString(),
+            BigInt(baseAsset.assetId),
+            liquidityAddedEvt.baseAmount,
+            pool.quoteAssetId,
+            liquidityAddedEvt.quoteAmount);
+
+        await ctx.store.save(pool);
+        await ctx.store.save(baseAsset);
+        await ctx.store.save(quoteAsset);
+        await ctx.store.save(tx);
+    } else {
+        throw new Error("Pool not found");
+    }
+}
+
+interface SwappedEvent {
+    poolId: bigint,
+    who: Uint8Array,
+    baseAsset: bigint,
+    quoteAsset: bigint,
+    baseAmount: bigint,
+    quoteAmount: bigint,
+    fee: bigint
+}
+
+function getSwappedEvent(event: PabloSwappedEvent): SwappedEvent {
+    if (event.isV2100) {
+        const {poolId, who, baseAsset, quoteAsset, baseAmount, quoteAmount, fee} = event.asV2100;
+        return {poolId, who, baseAsset, quoteAsset, baseAmount, quoteAmount, fee};
+    } else {
+        const {poolId, who, baseAsset, quoteAsset, baseAmount, quoteAmount, fee} = event.asLatest;
+        return {poolId, who, baseAsset, quoteAsset, baseAmount, quoteAmount, fee};
+    }
+}
+
+export async function processSwappedEvent(ctx: EventHandlerContext, event: PabloSwappedEvent) {
+    console.info('processing SwappedEvent', event);
+    const swappedEvt = getSwappedEvent(event);
+    const who = ss58.codec("picasso").encode(swappedEvt.who);
+    const pool = await get(ctx.store, PabloPool, swappedEvt.poolId.toString());
+    // only set values if the owner was missing, i.e a new pool
+    if (pool != undefined) {
+        const isReverse: boolean = pool.quoteAssetId != swappedEvt.quoteAsset;
+        const timestamp = BigInt(new Date().getTime());
+        pool.transactionCount += 1;
+        pool.calculatedTimestamp = timestamp;
+        pool.blockNumber = BigInt(ctx.block.height);
+        // find baseAsset: Following is only valid for dual asset pools
+        const baseAsset = pool.poolAssets
+            .find((asset) => asset.assetId != pool.quoteAssetId);
+        if (baseAsset == undefined) {
+            throw new Error('baseAsset not found');
+        }
+        // find quoteAsset
+        const quoteAsset = pool.poolAssets
+            .find((asset) => asset.assetId == pool.quoteAssetId);
+        if (quoteAsset == undefined) {
+            throw new Error('quoteAsset not found');
+        }
+        if (isReverse) {
+            console.debug('Reverse swap');
+            // volume
+            pool.totalVolume = Big(pool.totalVolume).add(Big(swappedEvt.baseAmount.toString())).toString();
```

Comment:
If I follow correctly, this total volume is not the one we are going to use to compute a volume in a timeframe (e.g. 24H)? If so, I assume you expect the querier to just extract the last 24h and do it themselves?

Reply:
yes the idea is that the party that queries can extract the volume at two timestamps(day_start, day_end) and then subtract to obtain the volume between the timestamps

Reply:
Sounds great :)